### PR TITLE
update jfrog credentials on change in gradle properties

### PR DESCRIPTION
**Pull request by stefancplace**
- update jfrog credentials on change in gradle properties
- read jfrog credentials from gradle.properties
- added unit/integration tests

### CODE REVIEW

**Comment by stefancplace** [comment]
For testing it is useful to clear the npm cache (`npm cache clean --force`), otherwise cplace packages might be taken from the local cache.
We should definitely test on Windows, could you help here @stophi-dev ?

File: src/model/RegistryInitializer.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    private getNpmrcPath(): string | undefined {
+        const npmConfig: string = execSync('npm config ls -l').toString();
+        const npmrcPath: string | undefined = (npmConfig.match(/userconfig *= *".*"/gi) || [])[0];
+        if (!npmrcPath) {
+            console.error(cred`✗`, 'No userconfig found in npm config');
+            return;
+        }
+        const cleanNpmrcPath: string = npmrcPath.replace(/^userconfig *= */, '').replace(/"/gi, '').replace(/\\\\/g, '\\');
+        if (!cleanNpmrcPath) {
+            console.error(cred`✗`, 'Userconfig was found in npmrc but but path can not be extracted');
```
Comment (Kaltenbach) [inline src/model/RegistryInitializer.ts:69]:
```suggestion
            console.error(cred`✗`, 'Userconfig was found in npmrc but path can not be extracted');
```

File: test/RegistryInitializer.test.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+import * as os from "os";
+
+const basePath = path.join(process.cwd(), 'testsetup');
+const gradleHome = path.join(basePath, RegistryInitializer.GRADLE_HOME);
+const gradlePropertiesPath = path.join(gradleHome, RegistryInitializer.GRADLE_PROPERTIES);
+const npmrcPath = path.join(basePath, '.npmrc');
+const npmrcUser = 'max.mustermann@collaboration-factory.de';
+const npmrcBasicAuthToken = 'bWF4Lm11c3Rlcm1hbm5AY29sbGFib3JhdGlvbi1mYWN0b3J5LmRlOnRva2Vu';
+
+const gradleProperties='org.gradle.java.home=/Users/stefanstadler/.sdkman/candidates/java/11.0.2-open\n' +
```
Comment (Kaltenbach) [inline test/RegistryInitializer.test.ts:15]: maybe without your local setup 😄

**Review by Kaltenbach — CHANGES_REQUESTED** [review]
LGTM - Only two minor things. Didn't test this locally...

**Review by Kaltenbach — APPROVED** [review]
👍

**Comment by Kaltenbach** [comment]
Just for me to understand: when would the `.npmrc` created? Neither in the old nor the new version it gets created but is expected to exist already, if I'm not mistaken 😄

**Comment by stefancplace** [comment]
The OS is asked upfront via "npm config ls -l" where .npmrc is stored (in userconfig key/value pair)
In case npm is not installed it should  -> console.error(cred`✗`, 'No user npmrc found');

**Comment by Kaltenbach** [comment]
Maybe s.th like a `touch ~/.npmrc` would be good enough - if that exists in the file sync lib.

**Review by Kaltenbach — APPROVED** [review]
Looks good. But still it would be nice to have this tested on Windows.

**Review by stophi-dev — APPROVED** [review]
Looks good. It also fixes the problem with older releases before gradle and therfore makes https://github.com/collaborationFactory/cplace-asc/pull/68 obsolete.

**Pull request merged**